### PR TITLE
Support zooming for arbitrary javafx.scene.chart.Axis implementations…

### PR DESCRIPTION
**Pull request by hwaite**
… (instead of just ValueAxis). This code was compiled against Java 8 but it should be trivial to make work with v7 if that's important.

### CODE REVIEW

File: jfxutils/src/main/java/org/gillius/jfxutils/chart/ChartZoomManager.java [hunk 1]
```diff
@@ -502,19 +527,112 @@ public void handle( ScrollEvent event ) {
...
+	}
+
+	private <T> DoubleProperty toDoubleProperty(Axis<T> axis, Property<T> property) {
+		final StringProperty stringProperty = new SimpleStringProperty();
+		final DoubleProperty doubleProperty = new SimpleDoubleProperty() {
+			// keep a reference so that the stringProperty doesn't get garbage-collected
+			private final StringProperty stringProp = stringProperty;
+		};
+
+		stringProperty.bindBidirectional(
```
Comment (gillius) [inline jfxutils/src/main/java/org/gillius/jfxutils/chart/ChartZoomManager.java]: Why do you need the intermediate string property when toNumericValue and toRealValue already convert to and from double? Why can't the doubleProperty bind directly to the property?

**Comment by gillius** [comment]
Thanks for your submission! I think the whole library is compatible with Java 7 for now but looking at the code I don't see anything specific to Java 8 here. I'm not sure Java 7 is a hard requirement; I simply started the library before JDK 8 was released so most of the code was written then.

What Axis types have lowerBoundProperty/upperBoundProperty besides ValueAxis? Or is this to support potential 3rd party Axis implementations?

**Comment by hwaite** [comment]
> What Axis types have lowerBoundProperty/upperBoundProperty besides ValueAxis? Or is this to support potential 3rd party Axis implementations?

I'd like to work with `extfx.scene.chart.DateAxis`.

**Comment by gillius** [comment]
OK, I see that extfx library and it makes sense. If others have similar issues in future jfxutils could be expanded to use lambda to find the property or have properties passed in directly, instead of using reflection and assumed method name.

The only question is whether or not that intermediate String property is needed, it looks like it converts the Date from DateAxis into a Double, then into a String, then back into a Double later... If it's not needed can you remove that extra step? If it is needed, that's fine, and I'll merge it as-is.

**Pull request merged**